scheduler/test/cook/test/mesos/mesos_mock.clj

```diff
@@ -0,0 +1,844 @@
+                                                     :command "10000"
+                                                     :custom-executor? false
+                                                     :memory mem
+                                                     :ncpus cpus))
+
+              user-b-job-ent-ids (for [_ (range 5)]
+                                   (create-dummy-job mesos-datomic-conn
+                                                     :user "b"
+                                                     :command "1000"
+                                                     :custom-executor? false
+                                                     :memory mem
+                                                     :ncpus cpus))
+              update-status-error-ms 200
+              ]
+          ;; Transact user "a"s jobs
+          (doall user-a-job-ent-ids)
+          ;; Let them get scheduled
+          (Thread/sleep 5000)
+          ;; Transact user "b"s jobs
+          (doall user-b-job-ent-ids)
+          ;; Let the system complete
+          (Thread/sleep 30000)
+          ;; Expect all jobs to run and complete, expect some of user "a"s jobs
+          ;; to be preempted
+          (let [job-ents (map (partial d/entity (d/db mesos-datomic-conn)) user-a-job-ent-ids)]
+            (is (some #(= (:reason/name (:instance/reason %)) :preempted-by-rebalancer) (mapcat :job/instance job-ents)))
+            (doseq [job-ent job-ents]
+              (let [instances (:job/instance job-ent)]
+                (is (= (:job/state job-ent) :job.state/completed))
+                (doseq [instance instances]
+                  (is (or (= (:instance/status instance) :instance.status/success)
+                          (= (:reason/name (:instance/reason instance)) :preempted-by-rebalancer)))
+                  (when (= (:instance/status instance) :instance.status/success)
+                    (is (< 0
+                           (- (.getTime (:instance/end-time instance))
+                              (.getTime (:instance/start-time instance))
+                              (read-string (:job/command job-ent)))
+                           update-status-error-ms)))))))
+          (doseq [job-ent-id user-b-job-ent-ids]
+            (let [job-ent (d/entity (d/db mesos-datomic-conn) job-ent-id)
+                  instances (:job/instance job-ent)
+                  instance (first instances)]
+              (is (= (count instances) 1))
+              (is (= (:job/state job-ent) :job.state/completed ))
+              (log/info "job i: " (d/touch job-ent))
+              (is (< 0
+                     (- (.getTime (:instance/end-time instance))
+                        (.getTime (:instance/start-time instance))
+                        (read-string (:job/command job-ent)))
+                     update-status-error-ms))))
+          (dump-jobs-to-csv (d/db mesos-datomic-conn) "trace.csv"))))))
```

Comment:
Yea, I added this because there was a bug with the integration and I couldn't figure it out without the trace. I will have it write only when log is set debug, rename, and add the ignore. 

Going to skip log/debug because I think it will look noisy in the log.